docs/source/developer/repomap.rst

```diff
@@ -0,0 +1,447 @@
+    future to improve organization.  Follow `issue #3807 <https://github.com/numba/numba/issues/3807>`_
+    for more details.
+
+
+Support Files
+-------------
+
+Build and Packaging
+'''''''''''''''''''
+
+- ``setup.py`` - Standard Python distutils/setuptools script
+- ``MANIFEST.in`` - Distutils packaging instructions
+- ``requirements.txt`` - Pip package requirements, not used by conda
+- ``versioneer.py`` - Handles automatic setting of version in installed package from git tags
+- ``.flake8`` - Preferences for code formatting.  Files should be fixed and removed from exception list as time allows.
+- ``buildscripts/condarecipe.local`` - Conda build recipe
+- ``buildscripts/remove_unwanted_files.py`` - Helper script to clean up files.  Used by build recipes.
+- ``buildscripts/condarecipe_clone_icc_rt`` - Recipe to build a standalone icc_rt package.  Not used anymore?
+
+
+Continuous Integration
+''''''''''''''''''''''
+- ``.binstar.yml`` - Binstar Build CI config (inactive)
+- ``azure-pipelines.yml`` - Azure Pipelines CI config (active: Win/Mac/Linux)
+- ``buildscripts/azure/`` - Azure Pipeline configuration for specific platforms
+- ``.travis.yml`` - Travis CI config (active: Mac/Linux, will be dropped in the future)
+- ``appveyor.yml`` - Appveyor CI config (inactive: Win)
+- ``buildscripts/appveyor/`` - Appveyor build scripts
+- ``buildscripts/incremental/`` - Generic scripts for building Numba on various CI systems
+- ``codecov.yml`` - Codecov.io coverage reporting
+
+
+Documentation / Examples
+''''''''''''''''''''''''
+- ``LICENSE`` - License for Numba
+- ``LICENSES.third-party`` - License for third party code vendored into Numba
+- ``README.rst`` - README for repo, also uploaded to PyPI
+- ``CONTRIBUTING.md`` - Documentation on how to contribute to project (out of date, should remove in favor of docs)
+- ``AUTHORS`` - List of Github users who have contributed PRs (out of date)
+- ``CHANGE_LOG`` - History of Numba releases, also directly embedded into Sphinx documentation
+- ``docs/`` - Documentation source
+- ``docs/_templates/`` - Directory for templates (to override defaults with Sphinx theme)
+- ``docs/Makefile`` - Used to build Sphinx docs with ``make``
+- ``docs/source`` - ReST source for Numba documentation
+- ``docs/_static/`` - Static CSS and image assets for Numba docs
+- ``docs/gh-pages.py`` - Utility script to update Numba docs (stored as gh-pages)
+- ``docs/make.bat`` - Not used (remove?)
+- ``examples/`` - Example scripts demonstrating numba (re/move to numba-examples repo?)
+- ``examples/notebooks/`` - Example notebooks (re/move to numba-examples repo?)
+- ``benchmarks/`` - Benchmark scripts (re/move to numba-examples repo?)
+- ``tutorials/`` - Tutorial notebooks (definitely out of date, should remove and direct to external tutorials)
+- ``numba/scripts/generate_lower_listing.py`` - Dump all registered lowering implementations for reference documentation
```

Comment:
IIRC this dumps everything which is `@lower*` decorated, so misses all the lowering that occurs via the extending API.